src/pydis_jam23/codecs/edges.py

```diff
@@ -0,0 +1,224 @@
+"""Encode a message into the given image.
+using as custom LSB variant
+
+One of the color channels is randomly seclected.
+On this channel edge detection is run.
+The pixels that are detected as edges will then be used,
+by to store data in the other two channels.
+
+Which channels is used is marked on the 0,0 pixel.
+Channel with a 1 as LSB is the mask
+"""
+from random import randint
+
+from PIL import Image, ImageDraw, ImageEnhance, ImageFilter
+
+from .common import CodecError, decode_varint, encode_varint
+
+cli_flag = "--edges"
+cli_help = "use the edges codec"
+
+
+def encode(image: Image.Image, message: bytes) -> None:
+    """encode a message into an image using the above described method"""
+    image_data = bytearray(image.tobytes())
+    data = encode_varint(len(message)) + message
+
+    mask_color = randint(0, 2)  # noqa: S311 no need for cryto
+    alternative_trys = 0
+
+    # check if the channels is big enugh for the message
+    # try the other ones if not
+    while True:
+        edges = get_edges(image.split()[mask_color]).tobytes()
+        if count_color(edges, 0) > len(data):  # count edge pixels
+            break
+        elif alternative_trys >= 2:
+            msg = "The message is to long to be encoded into this image."
+            raise CodecError(msg)
+        alternative_trys += 1
+        mask_color = (mask_color + alternative_trys) % 3
+
+    for idx in range(0, 3):  # set all LSB of the first image to 0
+        image_data[idx] = set_lsb(image_data[idx], 0)
```

Comment:
```suggestion
    for idx in range(0, 3):  # set all LSB of the first pixel to 0
        image_data[idx] = set_lsb(image_data[idx], 0)
```

1. What if the first pixel is detected as an edge, and this gets overwritten?
2. Needs adjusting if the image has more/less than 3 channels.

Reply:
by drawing a rectangle around the image i ensure that the border will never be detected as an edge, also i start data insertion at pixel 1 not 0